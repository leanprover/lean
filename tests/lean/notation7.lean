--
open nat

constant f : nat → nat
constant g : nat → nat → nat

notation A `:+1`:100000 := f A

#check g 0:+1:+1 (1:+1 + 2:+1):+1

set_option pp.notation false

#check g 0:+1:+1 (1:+1 + 2:+1):+1
