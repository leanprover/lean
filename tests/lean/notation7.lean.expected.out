g 0:+1:+1 (1:+1 + 2:+1):+1 : â„•
g (f (f 0)) (f (has_add.add (f 1) (f 2))) : nat
